src/robot/api/deco.py

```diff
@@ -14,9 +14,21 @@
 #  limitations under the License.
 
 import inspect
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Mapping,
+    Sequence,
+    Type,
+    TypeVar,
+    overload,
+)
 
+F = TypeVar("F", bound=Callable[..., Any])
```

Comment:
Since we can't use `ParamSpec` yet, I would prefer `Callable[..., Any]` since `mypy` does flag a plain `Callable`: `error: Missing type parameters for generic type "Callable"  [type-arg]`